File: extract-assets.py
#!/usr/bin/env python3
"""Extract CSS and JS from index.html into separate files"""

import os
import re

def extract_css_js():
    html_file = 'html/index.html'
    
    # Create directories
    os.makedirs('html/css', exist_ok=True)
    os.makedirs('html/js', exist_ok=True)
    
    print("Reading index.html...")
    with open(html_file, 'r', encoding='utf-8') as f:
        html_content = f.read()
    
    # Extract CSS content between <style> tags
    print("Extracting CSS...")
    css_match = re.search(r'<style>(.*?)</style>', html_content, re.DOTALL)
    if css_match:
        css_content = css_match.group(1).strip()
        with open('html/css/app.css', 'w', encoding='utf-8') as f:
            f.write(css_content)
        print(f"  ✓ Created html/css/app.css ({len(css_content)} chars)")
    
    # Extract console.js (first script block after body)
    print("Extracting console.js...")
    # Find the first script tag with the console override
    console_match = re.search(
        r'<script>\s*(\(function.*?onlineMode.*?\)\(\);)\s*</script>',
        html_content,
        re.DOTALL
    )
    if console_match:
        console_content = console_match.group(1).strip()
        with open('html/js/console.js', 'w', encoding='utf-8') as f:
            f.write(console_content)
        print(f"  ✓ Created html/js/console.js ({len(console_content)} chars)")
    
    # Extract main app.js (second script block)
    print("Extracting app.js...")
    # Find all script tags
    scripts = re.findall(r'<script>(.*?)</script>', html_content, re.DOTALL)
    if len(scripts) >= 2:
        app_js_content = scripts[1].strip()
        with open('html/js/app.js', 'w', encoding='utf-8') as f:
            f.write(app_js_content)
        print(f"  ✓ Created html/js/app.js ({len(app_js_content)} chars)")
    
    print("\n✓ Extraction complete!")

if __name__ == '__main__':
    extract_css_js()

File: extract-assets.js
#!/usr/bin/env node
const fs = require('fs');
const path = require('path');

// Create directories
const dirs = ['html/css', 'html/js'];
dirs.forEach(dir => {
   if (!fs.existsSync(dir)) {
        fs.mkdirSync(dir, { recursive: true });
    }
});

console.log('Reading index.html...');
const htmlFile = 'html/index.html';
const htmlContent = fs.readFileSync(htmlFile, 'utf-8');

// Extract CSS
console.log('Extracting CSS...');
const cssMatch = htmlContent.match(/<style>([\s\S]*?)<\/style>/);
if (cssMatch) {
    const cssContent = cssMatch[1];
    fs.writeFileSync('html/css/app.css', cssContent.trim());
    console.log(`  ✓ Created html/css/app.css (${cssContent.length} chars)`);
}

// Extract scripts
console.log('Extracting JavaScript...');
const scriptMatches = htmlContent.match(/<script>([\s\S]*?)<\/script>/g);

if (scriptMatches && scriptMatches.length >= 2) {
    // First script: console override
    const consoleScript = scriptMatches[0].match(/<script>([\s\S]*?)<\/script>/)[1];
    fs.writeFileSync('html/js/console.js', consoleScript.trim());
    console.log(`  ✓ Created html/js/console.js (${consoleScript.length} chars)`);
    
    // Second script: main app
    const appScript = scriptMatches[1].match(/<script>([\s\S]*?)<\/script>/)[1];
    fs.writeFileSync('html/js/app.js', appScript.trim());
    console.log(`  ✓ Created html/js/app.js (${appScript.length} chars)`);
}

console.log('\n✓ Extraction complete!');
console.log('\nNext steps:');
console.log('1. Update index.html to reference the new files');
console.log('2. Update service worker to cache CSS/JS files');
console.log('3. Minify CSS and JS files');
